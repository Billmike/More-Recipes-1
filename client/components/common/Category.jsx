import React from 'react';

/**
 * Category options component
 *
 * @returns {object} React element
 */
const Category = () => (
  <ul id="category" className="dropdown-content">
    <li><a href="#!" className="collection-item black-text">Breakfast</a></li>
    <li><a href="#!" className="collection-item black-text">Lunch</a></li>
    <li><a href="#!" className="collection-item black-text">Dinner</a></li>
    <li><a href="#!" className="collection-item black-text">Appetizer</a></li>
    <li><a href="#!" className="collection-item black-text">Main</a></li>
    <li><a href="#!" className="collection-item black-text">Dessert</a></li>
  </ul>
);

export default Category;
