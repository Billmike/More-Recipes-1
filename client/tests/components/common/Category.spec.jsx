import React from 'react';

import Category from '../../../components/common/Category';

const props = {
  handleSearchCategory: jest.fn(() => Promise.resolve()),
};

describe('Category component', () => {
  it('should render category component', () => {
    const wrapper = shallow(<Category {...props} />);
    expect(wrapper).toBeDefined();
    expect(wrapper.find('ul').length).toEqual(1);
    expect(wrapper.find('li').length).toEqual(6);
    expect(wrapper.find('ul').children().length).toEqual(6);
    expect(wrapper.find('ul').childAt(0).text()).toBe('Breakfast');
    expect(wrapper.find('ul').childAt(1).text()).toBe('Lunch');
    expect(wrapper.find('ul').childAt(2).text()).toBe('Dinner');
    expect(wrapper.find('ul').childAt(3).text()).toBe('Appetizer');
    expect(wrapper.find('ul').childAt(4).text()).toBe('Main');
    expect(wrapper.find('ul').childAt(5).text()).toBe('Dessert');
    expect(wrapper).toMatchSnapshot();
  });
  it('should search for a category', () => {
    const wrapper = shallow(<Category {...props} />);
    wrapper.find('[name="breakfast"]').simulate('click');
    expect(props.handleSearchCategory).toBeCalled();
    wrapper.find('[name="lunch"]').simulate('click');
    expect(props.handleSearchCategory).toBeCalled();
    wrapper.find('[name="dinner"]').simulate('click');
    expect(props.handleSearchCategory).toBeCalled();
    wrapper.find('[name="appetizer"]').simulate('click');
    expect(props.handleSearchCategory).toBeCalled();
    wrapper.find('[name="main"]').simulate('click');
    expect(props.handleSearchCategory).toBeCalled();
    wrapper.find('[name="dessert"]').simulate('click');
    expect(props.handleSearchCategory).toBeCalled();
    expect(wrapper).toMatchSnapshot();
  });
});
