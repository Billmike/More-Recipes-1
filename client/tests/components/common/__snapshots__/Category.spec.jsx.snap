// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category component should render category component 1`] = `
<ul
  className="dropdown-content"
  id="category"
>
  <li>
    <a
      className="collection-item black-text"
      name="breakfast"
      onClick={[MockFunction]}
      role="button"
      tabIndex="0"
    >
      Breakfast
    </a>
  </li>
  <li>
    <a
      className="collection-item black-text"
      name="lunch"
      onClick={[MockFunction]}
      role="button"
      tabIndex="0"
    >
      Lunch
    </a>
  </li>
  <li>
    <a
      className="collection-item black-text"
      name="dinner"
      onClick={[MockFunction]}
      role="button"
      tabIndex="0"
    >
      Dinner
    </a>
  </li>
  <li>
    <a
      className="collection-item black-text"
      name="appetizer"
      onClick={[MockFunction]}
      role="button"
      tabIndex="0"
    >
      Appetizer
    </a>
  </li>
  <li>
    <a
      className="collection-item black-text"
      name="main"
      onClick={[MockFunction]}
      role="button"
      tabIndex="0"
    >
      Main
    </a>
  </li>
  <li>
    <a
      className="collection-item black-text"
      name="dessert"
      onClick={[MockFunction]}
      role="button"
      tabIndex="0"
    >
      Dessert
    </a>
  </li>
</ul>
`;

exports[`Category component should search for a category 1`] = `
<ul
  className="dropdown-content"
  id="category"
>
  <li>
    <a
      className="collection-item black-text"
      name="breakfast"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
            Array [],
            Array [],
            Array [],
            Array [],
          ],
        }
      }
      role="button"
      tabIndex="0"
    >
      Breakfast
    </a>
  </li>
  <li>
    <a
      className="collection-item black-text"
      name="lunch"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
            Array [],
            Array [],
            Array [],
            Array [],
          ],
        }
      }
      role="button"
      tabIndex="0"
    >
      Lunch
    </a>
  </li>
  <li>
    <a
      className="collection-item black-text"
      name="dinner"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
            Array [],
            Array [],
            Array [],
            Array [],
          ],
        }
      }
      role="button"
      tabIndex="0"
    >
      Dinner
    </a>
  </li>
  <li>
    <a
      className="collection-item black-text"
      name="appetizer"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
            Array [],
            Array [],
            Array [],
            Array [],
          ],
        }
      }
      role="button"
      tabIndex="0"
    >
      Appetizer
    </a>
  </li>
  <li>
    <a
      className="collection-item black-text"
      name="main"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
            Array [],
            Array [],
            Array [],
            Array [],
          ],
        }
      }
      role="button"
      tabIndex="0"
    >
      Main
    </a>
  </li>
  <li>
    <a
      className="collection-item black-text"
      name="dessert"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
            Array [],
            Array [],
            Array [],
            Array [],
          ],
        }
      }
      role="button"
      tabIndex="0"
    >
      Dessert
    </a>
  </li>
</ul>
`;
